doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_NEXT_HOP_ID,
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator part of the SID
+     */
+    sai_uint8_t locator_len;
+
+    /**
+     * @brief Length of the Function part of the SID
+     */
+    sai_uint8_t function_len;
+
+    /**
+     * @brief Length of the Args part of the SID
+     */
+    sai_uint8_t args_len;
+
+    /**
+     * @brief Local Segment
+     */
+    sai_ip6_t sid;
+
+} sai_local_sid_entry_t;
+
+```
+
+5. Added apis to create/remove and set/get attributes for local_sid_entry
+
+    - add_local_sid_entry
+    - remove_local_sid_entry
+    - set_local_sid_entry_attribute
+    - get_local_sid_entry_attribute
+    - create_local_sid_entries
+    - remove_local_sid_entries
+    - set_local_sid_entries_attribute
+    - get_local_sid_entries_attribute
+
+
+## Examples ##
+- SR Headend
+    Example configuration for H.Encaps.Red behavior
+       
+        1. Create a SID list object with 3 segments
+
+            sidlist_entry_attrs[0].id = SAI_SEGMENTROUTE_SIDLIST_ATTR_TYPE
+            sidlist_entry_attrs01].value.s32 = SAI_SEGMENTROUTE_SIDLIST_TYPE_ENCAPS_RED
```

Comment:
Note: Adding Headend flavour within a SID-list will mean that same SID-list can not be used across different flavours.

Reply:
Agree. I wanted to split the Headend flavor from the SID list and add it as a separate attribute but left it there for backward compatibility reasons. I'm open to making this change now.

Reply:
will split it in the next PR.